python/mkdocs/docs/api.md

```diff
@@ -0,0 +1,336 @@
+<!--
+  - Licensed to the Apache Software Foundation (ASF) under one
+  - or more contributor license agreements.  See the NOTICE file
+  - distributed with this work for additional information
+  - regarding copyright ownership.  The ASF licenses this file
+  - to you under the Apache License, Version 2.0 (the
+  - "License"); you may not use this file except in compliance
+  - with the License.  You may obtain a copy of the License at
+  -
+  -   http://www.apache.org/licenses/LICENSE-2.0
+  -
+  - Unless required by applicable law or agreed to in writing,
+  - software distributed under the License is distributed on an
+  - "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  - KIND, either express or implied.  See the License for the
+  - specific language governing permissions and limitations
+  - under the License.
+  -->
+
+# Python API
+
+PyIceberg is based around catalogs to load tables. First, we need to instantiate a catalog. Let's say we have the following configuration
```

Comment:
Nit: use of "we" in docs. It's friendly, but docs tend to be more helpful if you stick to basic statements because you have to be more clear.

For example, rather than "First, we need to instantiate a catalog" you could say "The first step is to instantiate a catalog that can load tables." That gives more context about _why_ you'd load a catalog.